### ISSUE TITLE
[BUG] TorchLayer doesn't follow defined diff_method

### ISSUE THREAD

**Issue by OneRubberDucky** [PennyLaneAI/pennylane#4188]
### Expected behavior

There are two issues which are perhaps entangled:
1) the defined `diff_method` for `TorchLayer` does not follow the defined `diff_method`
2) the applied `diff_method` always falls back to either

- `parameter-shift` for PL `lightning`-enabled devices
- `backprop` for PL non-`lightning`-enabled devices

The following code was implemented on both GPU and non-GPU systems, and `lightning` and non-`lightning` enabled systems:

```
import numpy as np
import pennylane as qml
import torch
import sklearn.datasets

n_qubits = 2
#dev = qml.device("lightning.gpu", wires=n_qubits)
#dev = qml.device("lightning.qubit", wires=n_qubits)
dev = qml.device("default.qubit", wires=n_qubits)

#@qml.qnode(dev, diff_method = "backprop")
#@qml.qnode(dev, diff_method = "parameter-shift")
#@qml.qnode(dev, diff_method = "spsa")
#@qml.qnode(dev, diff_method = "finite-diff")
@qml.qnode(dev, diff_method = "adjoint")
def qnode(inputs, weights):
    qml.templates.AngleEmbedding(inputs, wires=range(n_qubits))
    qml.templates.StronglyEntanglingLayers(weights, wires=range(n_qubits))
    return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))

weight_shapes = {"weights": (3, n_qubits, 3)}

qlayer = qml.qnn.TorchLayer(qnode, weight_shapes)
clayer1 = torch.nn.Linear(2, 2)
clayer2 = torch.nn.Linear(2, 2)
softmax = torch.nn.Softmax(dim=1)
model = torch.nn.Sequential(clayer1, qlayer, clayer2, softmax)

print("the diff_method is")
print(qnode.best_method_str(dev, qnode.interface))

samples = 100
x, y = sklearn.datasets.make_moons(samples)
y_hot = np.zeros((samples, 2))
y_hot[np.arange(samples), y] = 1

X = torch.tensor(x).float()
Y = torch.tensor(y_hot).float()

opt = torch.optim.SGD(model.parameters(), lr=0.5)
loss = torch.nn.L1Loss()

epochs = 2
batch_size = 5
batches = samples // batch_size

data_loader = torch.utils.data.DataLoader(list(zip(X, Y)), batch_size=batch_size,
                                          shuffle=True, drop_last=True)

for epoch in range(epochs):

    running_loss = 0

    for x, y in data_loader:
        opt.zero_grad()

        loss_evaluated = loss(model(x), y)
        loss_evaluated.backward()

        opt.step()

        running_loss += loss_evaluated

    avg_loss = running_loss / batches
    print("Average loss over epoch {}: {:.4f}".format(epoch + 1, avg_loss))
```

When I cycle through the various `diff_methods` (e.g. `adjoint`, `spsa`, `finite-diff`, `parameter-shift`, `backprop`) and their various devices (e.g. `default.qubit`, `lightning.qubit`, `lightning.gpu`), I expect to get the defined `diff_method` (or error as appropriate). For example if I am using `diff_method = "adjoint"` I expect to be returned the correct `diff_method`, namely:

the diff_method is
adjoint

I also expect to be returned losses (the actual values may vary):
Average loss over epoch 1: 0.4996
Average loss over epoch 2: 0.3991

If an incorrect `device` and `diff_method` combination is defined, I expect to be thrown a warning and then a notification that my supplied arguments are falling back to certain defaults.

### Actual behavior

What I actually get is:

When I cycle through the various `diff_methods` (e.g. `adjoint`, `spsa`, `finite-diff`, `parameter-shift`, `backprop`) and their various devices (e.g. `default.qubit`, `lightning.qubit`, `lightning.gpu`), if I am using:

`default.qubit` I always get:
the diff_method is
backprop

`lightning.qubit` or `lightning.gpu` I always get:
the diff_method is
parameter-shift

I am then returned losses (as expected. The returned values may vary):
Average loss over epoch 1: 0.4996
Average loss over epoch 2: 0.3991

No warning is thrown when the applied `diff_method` defaults to certain settings.

### System information

```shell
`qml.about()` with non-GPU system:

Name: PennyLane
Version: 0.30.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml
Required-by: PennyLane-Lightning, PennyLane-Lightning-GPU

Platform info:           Linux-5.15.107+-x86_64-with-glibc2.31
Python version:          3.10.11
Numpy version:           1.22.4
Scipy version:           1.10.1
Installed devices:

`qml.about()` with GPU-enabled system:

Name: PennyLane
Version: 0.30.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml
Required-by: PennyLane-Lightning, PennyLane-Lightning-GPU

Platform info:           Linux-5.15.107+-x86_64-with-glibc2.31
Python version:          3.10.11
Numpy version:           1.22.4
Scipy version:           1.10.1
Installed devices:
- default.gaussian (PennyLane-0.30.0)
- default.mixed (PennyLane-0.30.0)
- default.qubit (PennyLane-0.30.0)
- default.qubit.autograd (PennyLane-0.30.0)
- default.qubit.jax (PennyLane-0.30.0)
- default.qubit.tf (PennyLane-0.30.0)
- default.qubit.torch (PennyLane-0.30.0)
- default.qutrit (PennyLane-0.30.0)
- null.qubit (PennyLane-0.30.0)
- lightning.qubit (PennyLane-Lightning-0.30.0)
- lightning.gpu (PennyLane-Lightning-GPU-0.30.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by OneRubberDucky** [comment]
Closing this issue since I just realized there was an error in my function call. Will reopen with an update if needed. Sorry about that!
